### PR TITLE
Auth updates & fix login as

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

This PR grabs some useful code from #5606 and a couple of other fixes related to authentication.
- Moves Google auth to EE (as it's a premium feature)
- Fixes a typo in Axes locked out view to clarify what view is for
- Moves logout view to views.py & removes extra unnecessary line
- Fixes logging as user (currently broken in production)

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: posthog/api/authentication.py [hunk 1]
```diff
@@ -11,7 +15,23 @@
...
 
-def axess_logout(*args, **kwargs):
+@csrf_protect
+def logout(request):
+    if request.user.is_authenticated:
+        request.user.temporary_token = None
+        request.user.save()
+
+    if is_impersonated_session(request):
+        restore_original_login(request)
```
Comment (paolodamico) [inline posthog/api/authentication.py:25]: Not sure if we should do this or completely kill the session. While this is nicer, I'm worried we're not properly handling cookies and distinct ID's when capturing data on PostHog and leading to unexpected merge users on our own team.

Comment (macobo) [reply posthog/api/authentication.py:25]: As long as we call `posthog.reset()` before `.identify` we should be fine or not sure what you mean.

Comment (paolodamico) [reply posthog/api/authentication.py:25]: You're right, we do!

File: posthog/api/authentication.py [hunk 2]
```diff
@@ -11,7 +15,23 @@
...
+        return redirect("/admin/")
+
+    response = auth_views.logout_then_login(request)
+    response.delete_cookie(settings.TOOLBAR_COOKIE_NAME, "/")
+
+    return response
+
+
+def axess_locked_out(*args, **kwargs):
```
Comment (Twixes) [inline posthog/api/authentication.py:34]: I think this should be a `axes_` not `axess_`? 😅

**Review by Twixes — COMMENTED** [review]
Overall looks good, except PR check failures, which don't actually seem valid 🤔

**Comment by paolodamico** [comment]
Hail Mary, will close this and open a separate PR, thanks everyone for the reviews.

**Pull request closed without merging**